lib/msal-common/src/client/DeviceCodeClient.ts

```diff
@@ -196,17 +197,23 @@ export class DeviceCodeClient extends BaseClient {
                 shrClaims: request.shrClaims,
                 sshKid: request.sshKid
             };
+
             const response = await this.executePostToTokenEndpoint(
                 this.authority.tokenEndpoint,
                 requestBody,
                 headers,
                 thumbprint);
 
-            if (response.body && response.body.error === Constants.AUTHORIZATION_PENDING) {
+            if (response.body && response.body.error) {
                 // user authorization is pending. Sleep for polling interval and try again
-                this.logger.info(response.body.error_description || "Authorization pending. Continue polling.");
-  
-                await TimeUtils.delay(pollingIntervalMilli);
+                if(response.body.error === Constants.AUTHORIZATION_PENDING) {
+                    this.logger.info(response.body.error_description || "Authorization pending. Continue polling.");
+                    await TimeUtils.delay(pollingIntervalMilli);
+                } else {
+                    // for any other error, throw
+                    this.logger.info("Unexpected error in polling from the server");
+                    throw ServerError.createUnexpectedError(response.body.error);
```

Comment:
Can we throw post_request_failed instead of a generic "unexpected" error? Like we do in msal-browser? This isn't necessarily a server error, is it?

Reply:
Yes. Any thing that fails : 4xx, 5xx etc will come under this + could be a router issues etc. I can add the new error type.

Reply:
Added a new error type.